base/repl/LineEdit.jl

```diff
@@ -1592,6 +1618,23 @@ buffer(s::PromptState) = s.input_buffer
 buffer(s::SearchState) = s.query_buffer
 buffer(s::PrefixSearchState) = s.response_buffer
 
+function empty_undo(s::PromptState)
+    empty!(s.undo_buffers)
+end
+empty_undo(s) = nothing
+
+function push_undo(s::PromptState)
```

Comment:
I would prefer the names of the new functions with an ending `!`, but it seems to be the style of the file, so better to keep as is for now.